lib/feature/browser/browser_tab_view/browser_tab_view.dart

```diff
@@ -0,0 +1,147 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_inappwebview/flutter_inappwebview.dart';
+
+class BrowserTabView extends StatefulWidget {
+  const BrowserTabView({required this.onScroll, super.key});
+
+  final void Function({
+    required int currentY,
+    required int gestureDY,
+  })? onScroll;
+
+  @override
+  State<BrowserTabView> createState() => _BrowserTabViewState();
+}
+
+class _BrowserTabViewState extends State<BrowserTabView> {
+  // Last SANE Y position (i.e. not overscrolled)
+  int? _lastScrollY;
+  // Position of gesture start: every direction change resets it
+  int? _scrollGestureYStart;
+  // Distance from gesture start to current position
+  int? _scrollGestureDY;
+  // Delayed scroll events are used to detect overscrolling
+  final List<DelayedScrollEvent> _delayedScrollEvents = [];
+  // How long to wait before considering scroll event as not overscroll
+  static const Duration _scrollTimerDelay = Duration(milliseconds: 100);
+
+  @override
+  Widget build(BuildContext context) {
+    return InAppWebView(
+      onOverScrolled: _onOverScrolled,
+      onScrollChanged: _onScrollChanged,
+      initialUrlRequest: URLRequest(
+        url: Uri.parse('https://app.flatqube.io/swap'),
+      ),
+    );
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+
+    for (final event in _delayedScrollEvents) {
+      event.timer.cancel();
+    }
+  }
+
+  void _onScrollChanged(_, __, int y) {
+    // Remove all events that are not active anymore (cancelled, executed)
+    _delayedScrollEvents.removeWhere((event) => !event.timer.isActive);
+
+    // New delayed scroll event, at this point we don't know if it's overscroll
+    final event = DelayedScrollEvent(
```

Comment:
As i know, InAppWebview has `pullToRefresh` callback so I don't understand why we need this

Reply:
Because overscroll is not ptr. We should filter all calls with overscrolled position for correctly working HUD.